calculator.c

```diff
@@ -0,0 +1,141 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+int numberStack[50];
+int nSTop = -1;
+int operatorStack[50];
+int opSTop = -1;
+
+int evaluateExp(char exp[]);
+void calculate();
+int operatorPrecedence(char operator);
+int custom_isdigit(char ch);
+int custom_isspace(char ch);
+
+int main()
+{
+    char exp[80];
+    int result = 0;
+    fgets(exp, sizeof(exp), stdin);
+    exp[strcspn(exp, "\n")] = '\0';
+    if (strlen(exp) == 0)
+    {
+        printf("Error: Invalid expression.\n");
+        return 0;
+    }
+    result = evaluateExp(exp);
+    printf("%d\n", result);
+    return 0;
+}
+
+int evaluateExp(char exp[])
```

Comment:
breakdown this function into smaller modular functions based on their responsibility.

Reply:
I agree with your suggestion to modularize the code further. I will refactor the evaluateExpression function by creating smaller functions to handle individual responsibilities like processing numbers, processing operators.